app/components-react/highlighter/ExportModal.tsx

```diff
@@ -0,0 +1,289 @@
+                ]}
+                value={privacy}
+                onChange={setPrivacy}
+              />
+            </Form>
+          </div>
+          <div
+            style={{
+              width: 300,
+              marginLeft: 24,
+              display: 'flex',
+              flexDirection: 'column',
+            }}
+          >
+            <video
+              src={v.exportInfo.file}
+              controls
+              style={{
+                width: '100%',
+                borderTopRightRadius: 8,
+                borderTopLeftRadius: 8,
+                outline: 'none',
+              }}
+            />
+            <div
+              style={{
+                borderBottomLeftRadius: 8,
+                borderBottomRightRadius: 8,
+                background: 'var(--section)',
+                fontWeight: 800,
+                padding: 12,
+              }}
+            >
+              {filename}
+              <br />
+              <a
+                onClick={() => {
+                  electron.remote.shell.showItemInFolder(v.exportInfo.file);
+                }}
+              >
+                Open file location
+              </a>
+            </div>
+          </div>
+        </div>
+        <div style={{ textAlign: 'right' }}>
+          <Button style={{ marginRight: 8 }} onClick={props.close}>
+            Close
+          </Button>
+          <Button
+            type="primary"
+            onClick={() => {
+              HighlighterService.actions.upload({
+                title,
+                description,
+                privacyStatus: privacy as TPrivacyStatus,
+              });
+            }}
+          >
+            Publish
+          </Button>
+        </div>
+      </div>
+    );
+  }
+
+  function getUploadProgress() {
+    return (
+      <div>
+        <h2>Upload Progress</h2>
+        <Progress
+          percent={Math.round((v.uploadInfo.uploadedBytes / v.uploadInfo.totalBytes) * 100)}
+          trailColor="var(--section)"
+          status={v.uploadInfo.cancelRequested ? 'exception' : 'normal'}
+        />
+        {!v.uploadInfo.cancelRequested && (
+          <div>
+            Uploading: {humanFileSize(v.uploadInfo.uploadedBytes, false)}/
+            {humanFileSize(v.uploadInfo.totalBytes, false)}
+          </div>
+        )}
+        {v.uploadInfo.cancelRequested && <span>Canceling...</span>}
+        <br />
+        <button
+          className="button button--soft-warning"
+          onClick={() => HighlighterService.actions.cancelUpload()}
+          style={{ marginTop: '16px' }}
+          disabled={v.uploadInfo.cancelRequested}
+        >
+          Cancel
+        </button>
+      </div>
+    );
+  }
+
+  function getUploadDone() {
+    const url = `https://youtube.com/watch?v=${v.uploadInfo.videoId}`;
+
+    return (
+      <div>
+        TODO
+        <br />
+        <a onClick={() => electron.remote.shell.openExternal(url)}>{url}</a>
+      </div>
+    );
+  }
```

Comment:
so a cool thing that React lets us do is define what in Vue would have been render functions like this to be full functional components that we can access with a much more readable `<MyComponent />` than a `getMyComponent()` not blocking at all but something to keep in mind as we move forward with React patterning

Reply:
Yeah for sure, good call out!  I've been really enjoying the ease of easily extracting sub-render-functions into separate components.  This one is a great candidate for splitting up as it's gotten pretty big.